rero_ils/modules/loans/api.py

```diff
@@ -98,53 +99,26 @@ def create(cls, data, id_=None, delete_pid=True,
             reindex=reindex, **kwargs)
         return record
 
-    def attach_item_ref(self):
-        """Attach item reference."""
-        item_pid = self.get('item_pid')
-        if not item_pid:
-            raise MissingRequiredParameterError(
-                description='item_pid missing from loan {0}'.format(
-                    self.pid))
-        self['item'] = self.loan_build_item_ref(item_pid, self)
-
-    def loan_build_item_ref(self, item_pid, loan):
-        """Build $ref for the Item attached to the Loan."""
-        base_url = current_app.config.get('RERO_ILS_APP_BASE_URL')
-        url_api = '{base_url}/api/{doc_type}/{pid}'
-        return {
-            '$ref': url_api.format(
-                base_url=base_url,
-                doc_type='items',
-                pid=item_pid)
-        }
-
-    def loan_build_patron_ref(self, patron_pid, loan):
-        """Build $ref for the Patron attached to the Loan."""
-        base_url = current_app.config.get('RERO_ILS_APP_BASE_URL')
-        url_api = '{base_url}/api/{doc_type}/{pid}'
-        return {
-            '$ref': url_api.format(
-                base_url=base_url,
-                doc_type='patrons',
-                pid=patron_pid)
-        }
-
-    def loan_build_document_ref(self, document_pid, loan):
-        """Build $ref for the Document attached to the Loan."""
-        base_url = current_app.config.get('RERO_ILS_APP_BASE_URL')
-        url_api = '{base_url}/api/{doc_type}/{pid}'
-        return {
-            '$ref': url_api.format(
-                base_url=base_url,
-                doc_type='documents',
-                pid=document_pid)
-        }
+    def date_fields2datetime(self):
+        """Convert string datetime fields to Python datetime."""
+        for field in self.DATE_FIELDS + self.DATETIME_FIELDS:
+            if field in self:
+                self[field] = str2datetime(self[field])
+
+    def date_fields2str(self):
+        """Convert Python datetime fields to string."""
+        for field in self.DATE_FIELDS:
+            if field in self:
+                self[field] = self[field].date().isoformat()
+        for field in self.DATETIME_FIELDS:
+            if field in self:
+                self[field] = self[field].isoformat()
```

Comment:
Maybe these 2 functions best places are into `modules/utils.py` ?

Reply:
These two functions need to be here for the invenio-circulation to work, sorry. it is related to the fact that our class `Loan(IlsRecord)` and not `Loan(Loan)` from invenio-circulation. We may be able to change this soon.